### ISSUE TITLE
Case Expressions

### ISSUE THREAD

**Issue by cpcloud** [ContinuumIO/blaze#1138]
Recap from the [mailing list](https://groups.google.com/a/continuum.io/forum/#!topic/blaze-dev/RB-UiCzYwco):

It'd be nice to be able to select values in a column based on values in other columns. SQL does this with [`case`](https://en.wikipedia.org/?title=SQL#Operators) expressions:

```sql
select a, case
    when a == 1 then b + 1
    when c == 2 then a
    else a
end
from table
```

SQLAlchemy implements this as well: http://docs.sqlalchemy.org/en/latest/core/sqlelement.html#sqlalchemy.sql.expression.case

NumPy has a special case of this (pun intended), namely `np.where`.

I propose that we do something similar in blaze. We need to preserve the order of the cases, in the case of overlapping conditions.

Proposed syntax:

```python
d = Data(..., dshape='var {a: int64, b: float64, c: int8}')
cased = case(
    (d.a == 1, d.b + 1),
    (d.c == 2, d.a),
    otherwise=d.a
)
r = transform(d, cased=cased)
```

**Comment by llllllllll** [comment]
In the `otherwise` case, should the `=` be a comma?

**Comment by cpcloud** [comment]
No, I was thinking it should be the only allowed keyword argument

**Comment by llllllllll** [comment]
Ah, that is pretty nice.
